src/main/java/seedu/address/logic/commands/EditTransactionCommand.java

```diff
@@ -0,0 +1,192 @@
+            + "are visible in the application!\n"
+            + "Use 'view' command to view a specific client before applying this command\n";
+
+    private final Index index;
+    private final EditTransactionDescriptor editTransactionDescriptor;
+
+    /**
+     * @param index of the transaction in the viewed client to edit.
+     * @param editTransactionDescriptor details to edit the transaction with.
+     */
+    public EditTransactionCommand(Index index, EditTransactionDescriptor editTransactionDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editTransactionDescriptor);
+
+        this.index = index;
+        this.editTransactionDescriptor = new EditTransactionDescriptor(editTransactionDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Client> lastShownList = model.getFilteredClientList();
+        if (lastShownList.size() != 1) {
+            throw new CommandException(MESSAGE_INVALID_USAGE);
+        }
+        Client currentClient = lastShownList.get(0);
+        TransactionLog transactionLog = currentClient.getTransactions();
+
+        if (index.getZeroBased() >= transactionLog.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToEdit = transactionLog.getTransaction(index.getZeroBased());
+        Transaction editedTransaction = createEditedTransaction(transactionToEdit, editTransactionDescriptor);
+        transactionLog.setTransaction(index.getZeroBased(), editedTransaction);
+
+        model.updateFilteredClientList(new NameEqualsKeywordPredicate(currentClient));
+        return new CommandResult(String.format(MESSAGE_EDIT_TRANSACTION_SUCCESS, editedTransaction));
+    }
+
+    /**
+     * Creates and returns a {@code Client} with the details of {@code clientToEdit}
+     * edited with {@code editClientDescriptor}.
+     */
+    private static Transaction createEditedTransaction(Transaction transactionToEdit,
+                                                       EditTransactionDescriptor editTransactionDescriptor) {
+        assert transactionToEdit != null;
+
+        Goods updatedGoods = editTransactionDescriptor.getGoods().orElse(transactionToEdit.getGoods());
+        Quantity updatedQuantity = editTransactionDescriptor.getQuantity().orElse(transactionToEdit.getQuantity());
+        Price updatedPrice = editTransactionDescriptor.getPrice().orElse(transactionToEdit.getPrice());
+        Date updatedDate = editTransactionDescriptor.getDate().orElse(transactionToEdit.getDate());
+
+        return transactionToEdit instanceof BuyTransaction
+                ? new BuyTransaction(updatedGoods, updatedPrice, updatedQuantity, updatedDate)
+                : new SellTransaction(updatedGoods, updatedPrice, updatedQuantity, updatedDate);
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Perhaps, it might be an improvement if this method is tested more thoroughly, as stated by codecov

Reply:
Thanks for pointing this out! I have added test cases for this method